CRM/Event/BAO/Event.php

```diff
@@ -2231,4 +2242,65 @@ public static function buildOptions($fieldName, $context = NULL, $props = array(
     return CRM_Core_PseudoConstant::get(__CLASS__, $fieldName, $params, $context);
   }
 
+  /**
+   * Change revenue recognition date for participants.
+   *
+   * @param array $event
+   */
+  public static function updateDeferredTransactions($event) {
+    if (empty($event['id'])) {
+      return;
+    }
+    $status = CRM_Contribute_PseudoConstant::contributionStatus();
+    $sql = "SELECT ft.*, cpp.contribution_id, eft.amount
+      FROM civicrm_participant_payment cpp
+      INNER JOIN civicrm_participant cp ON cp.id = cpp.participant_id
+      INNER JOIN civicrm_entity_financial_trxn eft ON eft.entity_id = cpp.contribution_id AND eft.entity_table = 'civicrm_contribution'
+      INNER JOIN civicrm_financial_trxn ft ON ft.id = eft.financial_trxn_id
+      INNER JOIN civicrm_entity_financial_account efa ON efa.financial_account_id = ft.from_financial_account_id
+      LEFT JOIN civicrm_option_group og ON og.name = 'account_relationship'
+      INNER JOIN civicrm_option_value ov ON ov.value = efa.account_relationship AND ov.option_group_id = og.id AND ov.name = 'Deferred Revenue Account is'
+      LEFT JOIN civicrm_option_group cog ON cog.name = 'contribution_status'
+      INNER JOIN civicrm_option_value cov ON cov.option_group_id = cog.id AND cov.name = 'Completed'
+      WHERE cp.event_id = %1 AND ft.from_financial_account_id = efa.financial_account_id AND ft.status_id = cov.value";
```

Comment:
Are we cancelling all the Deferred entries for contribution or just the latest entries?